packages/calcite-components/src/components/autocomplete/autocomplete.tsx

```diff
@@ -0,0 +1,845 @@
+    }
+
+    if (this.disabled) {
+      this.open = false;
+      return;
+    }
+
+    this.setFloatingElSize();
+    this.reposition(true);
+  }
+
+  private async documentClickHandler(event: MouseEvent): Promise<void> {
+    if (this.disabled || event.composedPath().includes(this.el)) {
+      return;
+    }
+
+    this.open = false;
+  }
+
+  private handleInternalAutocompleteItemSelect(event: Event): void {
+    this.value = (event.target as AutocompleteItem["el"]).value;
+    event.stopPropagation();
+    this.emitChange();
+  }
+
+  private mutationObserver = createObserver("mutation", () => this.getAllItemsDebounced());
+
+  private resizeObserver = createObserver("resize", () => {
+    this.setFloatingElSize();
+  });
+
+  onLabelClick(): void {
+    this.setFocus();
+  }
+
+  onBeforeOpen(): void {
+    this.calciteAutocompleteBeforeOpen.emit();
+  }
+
+  onOpen(): void {
+    this.calciteAutocompleteOpen.emit();
+  }
+
+  onBeforeClose(): void {
+    this.calciteAutocompleteBeforeClose.emit();
+  }
+
+  onClose(): void {
+    this.calciteAutocompleteClose.emit();
+  }
+
+  private emitChange(): void {
+    this.open = false;
+    this.calciteAutocompleteChange.emit();
+  }
+
+  private updateGroups(): void {
+    this.groups.forEach((group, index, items) => {
+      group.scale = this.scale;
+
+      if (index === 0) {
+        group.disableSpacing = true;
+      }
+
+      const nextGroupItem = items[index + 1];
+
+      if (nextGroupItem) {
+        nextGroupItem.disableSpacing = group.children.length === 0;
+      }
+    });
+  }
+
+  private updateItems(): void {
+    let activeDescendant: string = null;
+
+    this.items.forEach((item, index) => {
+      const isActive = index === this.activeIndex;
+
+      if (isActive) {
+        activeDescendant = item.guid;
+      }
+
+      item.active = isActive;
+      item.scale = this.scale;
+    });
+
+    this.activeDescendant = activeDescendant;
+  }
+
+  private handleInputFocus(): void {
+    this.open = true;
+  }
+
+  private handleContentTopSlotChange(event: Event): void {
+    this.hasContentTop = slotChangeHasAssignedElement(event);
+  }
+
+  private handleContentBottomSlotChange(event: Event): void {
+    this.hasContentBottom = slotChangeHasAssignedElement(event);
+  }
+
+  private getAllItemsDebounced = debounce(this.getAllItems, 0);
+
+  private getAllItems(): void {
+    const { el } = this;
+    this.groups = Array.from(el.querySelectorAll(groupItemSelector));
+    this.items = Array.from(el.querySelectorAll(itemSelector));
+    this.updateItems();
+    this.updateGroups();
+  }
+
+  private getIcon(): IconNameOrString {
+    const { icon } = this;
+
+    return icon === true ? ICONS.search : icon || ICONS.search;
+  }
+
+  private setReferenceEl(el: Input["el"]): void {
+    this.referenceEl = el;
+
+    if (!el) {
+      return;
+    }
+
+    this.resizeObserver?.observe(el);
+
+    connectFloatingUI(this);
+
+    // todo: fixme when supported in jsx
```

Comment:
Can you make the TODO uppercase so it's aligned with the pattern used during the lumina migration?